src/presentation/ui/page/mappings.templ

```diff
@@ -0,0 +1,364 @@
+			</div>
+		</template>
+		<template x-if="virtualHost.type == 'alias'">
+			<div class="mb-6 space-y-6">
+				@componentForm.InputField("text", "hostname", "Alias", "virtualHost.hostname", false)
+				@componentForm.InputField("text", "parentHostname", "Parent Hostname", "virtualHost.parentHostname", false)
+			</div>
+		</template>
+		@componentForm.BlockableSubmitButton("Create", "ph-check-fat", "closeCreateVhostModal", "isCreateVhostSubmitButtonBlocked", false)
+	</form>
+}
+
+templ CreateVhostModal() {
+	@componentStructural.Modal(
+		"Create Virtual Host", "isCreateVhostModalOpen", "closeCreateVhostModal()",
+		"",
+	) {
+		@CreateVhostForm()
+	}
+}
+
+func getVirtualHostHostnames(
+	vhostsWithMappings []dto.VirtualHostWithMappings,
+) []string {
+	vhostsHostnames := []string{}
+	for _, vhostWithMapping := range vhostsWithMappings {
+		vhostsHostnames = append(vhostsHostnames, vhostWithMapping.Hostname.String())
+	}
+
+	return vhostsHostnames
+}
+
+func getOnlyServiceNames() []string {
+	onlyServiceNames := []string{}
+	for serviceName := range valueObject.NativeSvcNamesWithAliases {
+		onlyServiceNames = append(onlyServiceNames, serviceName)
+	}
+
+	return onlyServiceNames
+}
+
+templ CreateMappingForm() {
+	<form
+		hx-post="/api/v1/vhosts/mapping/"
+		hx-indicator="#loading-overlay"
+		hx-swap="none"
+	>
+		<div class="-mt-4 my-6 space-y-6">
+			@componentMappings.MappingTypeSelectInput("targetType", "Type", "mapping.targetType", "mappingTypeShortDescription")
+			<template x-if="isCreateMappingFromVhost">
+				@componentForm.InputFieldReadOnly("text", "hostname", "Hostname", "virtualHost.hostname", false)
+			</template>
+			<template x-if="!isCreateMappingFromVhost">
+				@componentForm.SelectInputFromAlpine("hostname", "Hostname", "virtualHost.hostname", "availableVhostHostnames", true, false)
+			</template>
+			@componentForm.InputFieldWithPrefix("text", "path", "Source URL", "mapping.path", "vhostHostnameWithTrailingSlash", false)
+		</div>
+		<template x-if="mapping.targetType == 'service'">
+			<div class="mb-6 space-y-6">
+				@componentForm.SelectInput("targetValue", "Service Name", "mapping.targetValue", getOnlyServiceNames(), true, false)
+			</div>
+		</template>
+		<template x-if="mapping.targetType == 'url'">
+			<div class="mb-6 space-y-6">
+				@componentForm.InputField("text", "targetValue", "Destination URL", "mapping.targetValue", false)
+			</div>
+		</template>
+		<template x-if="mapping.targetType == 'inline-html'">
+			<div class="mb-6 space-y-6">
+				@componentForm.TextArea("targetValue", "Inline HTML", "mapping.targetValue", false)
+				@componentMappings.ResponseCodeSelectInput("targetHttpResponseCode", "Response Code", "mapping.targetHttpResponseCode")
+			</div>
+		</template>
+		<template x-if="mapping.targetType == 'response-code'">
+			<div class="mb-6 space-y-6">
+				@componentMappings.ResponseCodeSelectInput("targetValue", "Response Code", "mapping.targetValue")
```

Comment:
I left it as `targetValue`, unlike in `inline-html`, because all mappings use `targetValue` for their main values. I thought this convention should also apply to `response-code` mappings, while in `inline-html`, the status is a secondary input rather than a primary one.